nixos/modules/services/web-apps/ackee.nix

```diff
@@ -0,0 +1,131 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ackee;
+  defaultSettings = {
+    ACKEE_MONGODB = "mongodb://localhost:27017/ackee";
+    ACKEE_USERNAME = "ackee";
+    ACKEE_PORT = "3000";
+    NODE_ENV = "production";
+    HOME = "%S/ackee";
+  };
+in
+{
+  options.services.ackee = {
+    enable = mkEnableOption "the Ackee analytics tool";
+
+    settings = mkOption {
+      type = with types; attrsOf (nullOr (oneOf [ str path package ]));
```

Comment:
Out of curiosity what does this represent?
```
services.ackee.settings.foo = null;
```

Reply:
I guess it could be used to unset one of the default environment variables, but that doesn’t make much sense, so I removed the `nullOr`.